### PR TITLE
add default navmesh construction config options to SimulatorConfig

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

After #1351 we want config hooks for changing the default navmesh construction behavior for the simulator.

This PR adds the option to turn off default navmesh and to include static objects. Defaults are set to current behavior.  

This represents the first step. Ideally we expand these options further with an additional dataclass or other mechanism for detailed navmesh parameter tuning from config.

## How Has This Been Tested

Not tested yet.

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

**Review by mukulkhanna — APPROVED** [review]
LGTM!

File: habitat-lab/habitat/config/default_structured_configs.py [hunk 1]
```diff
@@ -1442,6 +1442,12 @@ class SimulatorConfig(HabitatBaseConfig):
     # If the number of agents is greater than one,
     # then agents_order has to be set explicitly.
     agents_order: List[str] = MISSING
+
+    # Simulator should use default navmesh settings from agent config
+    default_agent_navmesh: bool = True
```
Comment (wangyenjen) [inline habitat-lab/habitat/config/default_structured_configs.py:1447]: Hi @aclegg3, thank you for the excellent series of refactoring!

I completely understand the importance of this motivation. However, I would like to suggest setting `default_agent_navmesh` to `False` until the new feature has undergone thorough testing. Currently, when running `python examples/example.py`, there are some additional error messages that appear, such as `AttributeError: 'habitat_sim._ext .habitat_sim_bindings.SimulatorConfiguration' object has no attribute 'navmesh_settings'`.

Comment (aclegg3) [reply habitat-lab/habitat/config/default_structured_configs.py:1447]: @wangyenjen thanks for the feedback. 

From the error it looks like you may not have updated locally to habitat-sim `main`. That attribute was added in habitat-sim [2111](https://github.com/facebookresearch/habitat-sim/pull/2111) and habitat-lab [1351](https://github.com/facebookresearch/habitat-lab/pull/1351). Try again with nightly conda build or source build on main. :) 

Note that this behavior (`default_agent_navmesh=True`) mimics the expected behavior from before the above PRs.

Comment (wangyenjen) [reply habitat-lab/habitat/config/default_structured_configs.py:1447]: @aclegg3, I understand. In fact, I also tried the nightly conda build for the main branch but encountered the same error message. Perhaps I should attempt to build the current main branch for testing purposes. Thank you very much!

**Review by vincentpierre — APPROVED** [review]

**Pull request merged**